src/main/g8/src/main/java/com/example/helloworld/GreeterServer.java

```diff
@@ -0,0 +1,122 @@
+package com.example.helloworld;
+
+//#import
+
+import akka.actor.ActorSystem;
+import akka.http.javadsl.*;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.HttpResponse;
+import akka.japi.Function;
+import akka.stream.ActorMaterializer;
+import akka.stream.Materializer;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyFactory;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.SecureRandom;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Base64;
+import java.util.concurrent.CompletionStage;
+
+
+//#import
+
+//#server
+class GreeterServer {
+
+  static void main(String[] args) throws Exception {
+    // important to enable HTTP/2 in ActorSystem's config
+    Config conf = ConfigFactory.parseString("akka.http.server.preview.enable-http2 = on")
+      .withFallback(ConfigFactory.defaultApplication());
+    ActorSystem system = ActorSystem.create("HelloWorld", conf);
+    new GreeterServer(system).run();
+  }
+
+  final ActorSystem system;
+
+  public GreeterServer(ActorSystem system) {
+    this.system = system;
+  }
+
+  public CompletionStage<ServerBinding> run() throws Exception {
+
+    Materializer materializer = ActorMaterializer.create(system);
+
+    Function<HttpRequest, CompletionStage<HttpResponse>> service =
+        GreeterServiceHandlerFactory.create(
+            new GreeterServiceImpl(materializer),
+            materializer);
+
+    CompletionStage<ServerBinding> bound =
+        Http.get(system).bindAndHandleAsync(
+          service,
+          ConnectWithHttps.toHostHttps("127.0.0.1", 8080)
+              .withCustomHttpsContext(serverHttpContext()),
+          materializer
+        );
+
+    bound.thenAccept(binding ->
+        System.out.println("gRPC server bound to: " + binding.localAddress())
+    );
+
+    return bound;
+  }
+  // #server
+
+
+  // FIXME this will be replaced by a more convenient utility, see https://github.com/akka/akka-grpc/issues/89
```

Comment:
hmm not sure we figured out yet what to do here, but we can leave that for another PR